### ISSUE TITLE
CKEditor 4 not recognized by WYSIWYG

### ISSUE THREAD

**Issue by leolandotan** [promet/drupal7-framework#14]
After doing a fresh install the WYSIWYG module can't detect the version of the CKEditor library due to an [issue](https://www.drupal.org/node/1853550) in Drupal.org. This causes the Promet Default text format features to not be successfully reverted.

My initial solution is to
* Create a patch using the [netresearch/composer-patches-plugin](https://github.com/netresearch/composer-patches-plugin) and adding it into the **repositories** array.

```
{
    "type": "package",
    "package": {
        "name": "promet/drupal7-framework-patches",
        "version": "1.0.0",
        "type": "patches",
        "require": {
            "netresearch/composer-patches-plugin": "~1.0",
            "drupal/wysiwyg": "7.2.2"
        },
        "extra": {
            "patches": {
                "drupal/wysiwyg": {
                  "7.2.2": [
                        {
                            "title": "Ckeditor 4.0 - The version of CKEditor could not be detected.",
                            "url": "https://www.drupal.org/files/wysiwyg-ckeditor4-1853550-4.patch"
                        }
                    ]
                }
            }
        }
    }
}
```

* Include the "patch package" into the **require** array `"promet/drupal7-framework-patches": "~1.0.0",`
* Remove the `"drupal/wysiwyg": "7.*",` from the **require** array since we have declared it inside the "patch package".
* Then do `composer update`

Any thoughts on this issue and solution or should we focus on [Replace the wysiwyg module by the ckeditor module ](https://github.com/promet/drupal7-framework/issues/10)?

**Comment by Luukyb** [comment]
Thank you for that ! Was this solution you posted working on your end with a fresh install ?
I think in the future we should use the ckeditor by default, and get rid of the WYSIWYG.

**Comment by leolandotan** [comment]
You're welcome! Yes it did. After a success on `composer update` I destroyed the VM and did `vagrant up --provision` again.

**Comment by Luukyb** [comment]
@leolandotan if your solution is working, could you do a PR for it ? Thanks

**Comment by leolandotan** [comment]
@Luukyb Ok but once I try to do an ```composer update``` I get the following error.
```
Loading composer repositories with package information
Updating dependencies (including require-dev)            
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - fabpot/goutte v1.0.7 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - fabpot/goutte v1.0.6 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - fabpot/goutte v1.0.5 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - fabpot/goutte v1.0.4 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - fabpot/goutte v1.0.3 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - fabpot/goutte v1.0.2 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - fabpot/goutte v1.0.1 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - behat/mink-goutte-driver v1.1.0 requires behat/mink ~1.6@dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.7 requires behat/mink ~1.4.3 -> no matching package found.
    - behat/mink-goutte-driver v1.0.6 requires behat/mink >=1.4,<1.5.0-dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.5 requires behat/mink >=1.4,<1.5.0-dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.4 requires behat/mink >=1.4,<1.5.0-dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.3 requires behat/mink >=1.4.0,<1.5.0-dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.2 requires behat/mink >=1.4.0,<1.5.0-dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.1 requires behat/mink >=1.4.0,<1.5.0-dev -> no matching package found.
    - behat/mink-goutte-driver v1.0.0 requires behat/mink >=1.4.0,<1.5.0-dev -> no matching package found.
    - drupal/drupal-extension v0.1.5 requires behat/mink-goutte-driver * -> satisfiable by behat/mink-goutte-driver[v1.0.0, v1.0.1, v1.0.2, v1.0.3, v1.0.4, v1.0.5, v1.0.6, v1.0.7, v1.0.8, v1.0.9, v1.1.0].
    - behat/mink-goutte-driver v1.0.8 requires fabpot/goutte ~1.0.1 -> satisfiable by fabpot/goutte[v1.0.1, v1.0.2, v1.0.3, v1.0.4, v1.0.5, v1.0.6, v1.0.7].
    - behat/mink-goutte-driver v1.0.9 requires fabpot/goutte ~1.0.1 -> satisfiable by fabpot/goutte[v1.0.1, v1.0.2, v1.0.3, v1.0.4, v1.0.5, v1.0.6, v1.0.7].
    - Installation request for drupal/drupal-extension 0.1.5 -> satisfiable by drupal/drupal-extension[v0.1.5].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```
Any thoughts on this?

**Comment by Luukyb** [comment]
I am not using the framework d7 much. Maybe @generalredneck or @dsdeiz would know ?

**Comment by leolandotan** [comment]
@Luukyb I really forgot to update about the `composer update` issue. I think what I did was run the command inside vagrant or installed curl using `sudo apt-get install php5-curl`

**Comment by dsdobrzynski** [comment]
Since the stable version of the WYSIWYG is really old, we've been using the dev version with the full/stable version of the CKEditor library on projects. Resolved in [PR #17](https://github.com/promet/drupal7-framework/pull/17)

The composer problem is no longer an issue.

Closing this.
